packages/contracts-bedrock/contracts/L1/SystemConfig.sol

```diff
@@ -207,4 +214,15 @@ contract SystemConfig is OwnableUpgradeable, Semver {
             sstore(slot, _unsafeBlockSigner)
         }
     }
+
+    /**
+     * @notice Returns the minimum gas limit allowed by the system. If the L2
+     *         gas limit is set to a value smaller than this, then it is
+     *         possible for a block to be produced that uses more gas than what
+     *         is allowed on L2, resulting in a liveliness failure.
+     */
+    function _minimumGasLimit() internal returns (uint256) {
+        uint256 maxResourceLimit = uint256(ResourceMetering(PORTAL).MAX_RESOURCE_LIMIT());
+        return maxResourceLimit + SYSTEM_TRANSACTION_MAX_GAS;
```

Comment:
Instead of reading from the constant from the portal contract, we can also define it here, and read the system config from the resource-metering (or really from the optimism portal) contract. This prevents a circular dependency later on if we end up making the system-config a dependency of the portal contract to read other system configurables.

Reply:
Thats a good point, the call overhead wont matter because its in the gas burn path and the gas used in computing the eip1559 params counts towards the gas burning

Reply:
After thinking about this a bit, i agree that its a good idea to centralize as many config params in a single place for governance purposes, then do we move the eip 1559 params there too? Due to needing to make a lot of decisions, i lean towards not making any drastic changes as part of this pr